lib/util.js

```diff
@@ -184,10 +186,15 @@ function getConstructorOf(obj) {
 
 
 function inspectPromise(p) {
-  Debug = Debug || require('vm').runInDebugContext('Debug');
-  var mirror = Debug.MakeMirror(p, true);
-  if (!mirror.isPromise())
+  if (!Debug) {
+    const runInDebugContext = require('vm').runInDebugContext;
+    const result = runInDebugContext('[Debug, ObjectIsPromise]');
+    Debug = result[0];
+    ObjectIsPromise = result[1];
+  }
```

Comment:
To prepare the ground for the use of `MakeMirror` in https://github.com/nodejs/node/pull/3119, may I suggest to move this to a separate function ? `setupDebugObjects()` for example.